dcos/subcommand.py

```diff
@@ -435,3 +465,77 @@ def package_json(self):
         package_json_path = os.path.join(self._dir(), 'package.json')
         with util.open_file(package_json_path) as package_json_file:
             return util.load_json(package_json_file)
+
+
+class SubcommandMain():
+    """Representes a subcommand running in the main thread"""
+
+    def __init__(self, command):
+        """
+        :param commad: function to run in thread
+        :type command: func
+        """
+
+        self.command = command
+
+    def run_and_capture(self):
+        """
+        Run a command and capture exceptions. This is a blocking call
+        :returns: tuple of exitcode, error (or None)
+        :rtype: int, str | None
+        """
+
+        command_module = "dcoscli.{}.main".format(self.command)
+        m = import_module(command_module)
```

Comment:
I am not super excited about this pattern. We already have a static list of default subcommands. Why not just turn that into a map/dictionary? We can use the map here to find the function. If the map doesn't have the key then we know that we need to exec a process.